File: 8kyu/(8 kyu) Remove First and Last Character/(8 kyu) Remove First and Last Character.go
package kata

// 1 Simple solution
func RemoveChar(word string) string {
	result := ""
	for i := 1; i < len(word)-1; i++ {
		result += string(word[i])
	}
	return result
}

// 2 Optimized solution
func RemoveCharOptimized(word string) string {
	return word[1 : len(word)-1]
}

File: 8kyu/(8 kyu) Remove First and Last Character/(8 kyu) Remove First and Last Character.rs
// 1 Simple solution
pub fn remove_char(s: &str) -> String {
    let mut res = Vec::new();
    for (i, c) in s.chars().enumerate() {
        if i != 0 && i != s.len() - 1 {
            res.push(c);
        }
    }
    String::from_iter(res)
}

// 2 Optimized solution
pub fn remove_char(s: &str) -> String {
    s.chars().skip(1).take(s.len() - 2).collect()
}

// 3 Clever solution
pub fn remove_char(s: &str) -> String {
    s[1..s.len() - 1].to_string()
}
